src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -38,87 +39,93 @@ public interface IConcurrencyManager
             numberOfCoresToUse = _maximumNumberOfCoresToUse;
+
          var concurrentData = new ConcurrentQueue<TData>(data);
          numberOfCoresToUse = Math.Min(numberOfCoresToUse, concurrentData.Count);
 
          var results = new ConcurrentDictionary<TData, ConcurrencyManagerResult<TResult>>();
-         //Starts one task per core
-         var tasks = Enumerable.Range(0, numberOfCoresToUse).Select(async coreIndex =>
+
+         //Splits the action based in the number of cores available. 
+         //No thread will be created here. If the actions are all running on the same thread, the effect of the concurrency execution will be inexistent
+
+         var tasks = Enumerable.Range(0, numberOfCoresToUse).Select(async coreId =>
          {
             //While there is data left
             while (concurrentData.TryDequeue(out var datum))
             {
                cancellationToken.ThrowIfCancellationRequested();
 
-               //Invoke the action on it and store the result
+               //Invoke the action on it and store the result. We assume here that each action runs on its own thread.
                var result = await returnWithExceptionHandling(
-                  coreIndex,
-                  cancellationToken,
-                  action, 
+                  coreId,
+                  action,
                   datum,
-                  id
+                  idFunc,
+                  cancellationToken
                );
                results.TryAdd(datum, result);
             }
-         }).ToList();
+         });
 
          await Task.WhenAll(tasks);
-         //all tasks are completed. Can return results
 
-         var tt = results.Values;
+         //all tasks are completed. Can return results
          return results;
       }
 
       private async Task<ConcurrencyManagerResult<TResult>> returnWithExceptionHandling<TData, TResult>
       (
-         int coreId, 
-         CancellationToken cancellationToken, 
-         Func<int, CancellationToken, TData, Task<TResult>> task, 
-         TData input, Func<TData, string> id
-      ) where TResult : class
+         int coreId,
+         Func<int, TData, CancellationToken, Task<TResult>> task,
+         TData data,
+         Func<TData, string> idFunc,
+         CancellationToken cancellationToken
+      )
       {
          try
          {
             return new ConcurrencyManagerResult<TResult>
             (
-               id(input),
-               result: await task.Invoke(coreId, cancellationToken, input)
+               id: idFunc(data),
+               result: await task(coreId, data, cancellationToken)
             );
          }
          catch (Exception e)
          {
             return new ConcurrencyManagerResult<TResult>
             (
-               id(input),
-               errorMessage: e.Message
+               id: idFunc(data),
+               errorMessage: e.ExceptionMessage(addContactSupportInfo: false)
```

Comment:
when you run stuff in //, sometimes you need to aggregate deeper to see the error. The ExceptionMessage extension method does it for you